source/extensions/filters/http/grpc_web/grpc_web_filter.cc

```diff
@@ -30,27 +63,17 @@ struct RcDetailsValues {
 };
 using RcDetails = ConstSingleton<RcDetailsValues>;
 
-// Bit mask denotes a trailers frame of gRPC-Web.
-const uint8_t GrpcWebFilter::GRPC_WEB_TRAILER = 0b10000000;
-
-// Supported gRPC-Web content-types.
-const absl::flat_hash_set<std::string>& GrpcWebFilter::gRpcWebContentTypes() const {
-  static const absl::flat_hash_set<std::string>* types = new absl::flat_hash_set<std::string>(
-      {Http::Headers::get().ContentTypeValues.GrpcWeb,
-       Http::Headers::get().ContentTypeValues.GrpcWebProto,
-       Http::Headers::get().ContentTypeValues.GrpcWebText,
-       Http::Headers::get().ContentTypeValues.GrpcWebTextProto});
-  return *types;
-}
-
-bool GrpcWebFilter::isGrpcWebRequest(const Http::RequestHeaderMap& headers) {
-  if (!headers.Path()) {
-    return false;
-  }
-  const Http::HeaderEntry* content_type = headers.ContentType();
-  if (content_type != nullptr) {
-    return gRpcWebContentTypes().count(content_type->value().getStringView()) > 0;
+bool GrpcWebFilter::checkGrpcResponseHeaders(Http::ResponseHeaderMap& headers, bool end_stream) {
+  const bool ok = Http::Utility::getResponseStatus(headers) == enumToInt(Http::Code::OK);
+  if (ok && (Grpc::Common::isGrpcResponseHeaders(headers, end_stream) ||
+             hasGrpcWebContentType(headers))) {
+    return true;
   }
+
+  ENVOY_LOG(info, "upstream response is not gRPC response");
+  headers.setGrpcStatus(
+      Grpc::Utility::httpToGrpcStatus(enumToInt(Http::Utility::getResponseStatus(headers))));
+  response_headers_ = &headers;
```

Comment:
Since we mutate the response header I would change the name of this function to reflect that